batch.go

```diff
@@ -281,10 +281,13 @@ func (b *Batch) die(err error) {
 	}
 
 	b.err = err
-	b.conn.die(err)
+	if b.conn != nil {
+		err = b.conn.termContext(err)
```

Comment:
This prevents a panic when trying to release a connection during `Close()` when that connection that was killed without terminating the context.